src/NHibernate.Test/NHSpecificTest/GH1149/Mappings.hbm.xml

```diff
@@ -0,0 +1,25 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<hibernate-mapping xmlns="urn:nhibernate-mapping-2.2"
+										namespace="NHibernate.Test.NHSpecificTest.GH1149"
+										assembly="NHibernate.Test">
+	<class name="Company" >
+		<id name="Id">
+			<generator class="native"/>
+		</id>
+		<property name="Name" />
+
+		<one-to-one name="Address" class="Address" property-ref="Company" cascade="all-delete-orphan"/>
```

Comment:
May you add a test with an actual `one-to-one` in db? I mean, a `one-to-one` not using a `property-ref` but with one entity pk being also the pk of the other entity (so a pk which is also a fk), both side mapped as `one-to-one` with one side being `inverse`. (`PersonTrueO2O` with `AddressTrueO2O` in NH3931 tests is an example of this.)

Unless this case is already tested in NHibernate.Test and working (meaning the issue report was not really precise).

Reply:
Done, PR updated.